src/dbnode/persist/fs/files.go

```diff
@@ -855,8 +856,9 @@ func ReadIndexInfoFiles(
 			var info index.IndexVolumeInfo
 			err := info.Unmarshal(data)
 			infoFileResults = append(infoFileResults, ReadIndexInfoFileResult{
-				ID:   id,
-				Info: info,
+				ID:         id,
+				Info:       info,
+				PathPrefix: fmt.Sprintf("%s*", pathPrefixFromIndexInfoFilePath(filepath)),
```

Comment:
Might be able to use `FileSetFile` if we modify forEachInfoFile to return that instead of filepath/id.